pkgs/servers/homepage-dashboard/default.nix

```diff
@@ -9,8 +9,23 @@
 , fetchpatch
 , makeBinaryWrapper
 , nixosTests
+, enableLocalIcons ? true
 }:
+let
+  dashboardIcons = fetchFromGitHub {
+    owner = "walkxcode";
+    repo = "dashboard-icons";
+    rev = "a02a5999fe56948671721da8b0830cdd5b609ed7"; # Until 2024-02-25
+    hash = "sha256-s0Doh4j6CH66fZoQKMt4yc7aLStNFGMVoDp5dvs7+pk=";
+  };
 
+  installLocalIcons = ''
+    mkdir -p $out/share/homepage/public/icons
+    cp ${dashboardIcons}/png/* $out/share/homepage/public/icons
+    cp ${dashboardIcons}/svg/* $out/share/homepage/public/icons
+    cp ${dashboardIcons}/LICENSE $out/share/homepage/public/icons/
```

Comment:
So home-dashboard checks these local paths before generating URLs to the dashboard-icons repo on github (which requires internet access)? No other config needed?

Reply:
Related, do you have any thoughts on how to add custom icons/images (also for offlline use)?

It seems upstream wants users to bind mount the images/icons over the home-dashboard publlic/icons (or images?) directory. I guess an option for that could be added to the NixOS module, but if it can be done at the derivation/package level I think that's better.

Some ideas I have for package parameters:
* `extraAssets ? []` -- list of derivations that contain `$out/share/homepage/public/{icons,images}` that get symlinked into homepage-dashboard $out?
* `extraIcons ? {}, extraImages ? {}` -- attrset of (basename_of_asset, path_to_asset)

Or maybe we don't need anything on nixpkgs side and users can `homepage-dashboard.overrideAttrs (old: { postInstall = (old.postInstall or "") + "...copy_extra_files..."; })`?

However, al of the above options require re-building homepage-dashboard. For injecting custom icons/images, a two-derivation configuration would be nice.

Reply:
No, '/icons/' needs to be added to the icon path. Since it is hard to add local icons after installation, maybe we can integrate icons into the package.

Reply:
> No, '/icons/' needs to be added to the icon path. 

I think it would be good to add a comment about that in the Nix expressions. Because I did not realize I need to make NixOS config changes to benefit from this PR.

Perhaps even a NixOS changelog entry, if you like.

Reply:
Can we make symlinks instead of copying?

I was thinking of adding extra icons, but each time I'd make a change there would be a new ~100 MiB store path. At least with symlinking, each path would be "only" 55 MiB (homepage-dashboard itself, without the local icons).